tests/app/questionnaire/test_router.py

```diff
@@ -506,6 +506,81 @@ def test_section_summary_on_completion_false(self):
 
         assert expected_location_url == next_location
 
+    @pytest.mark.usefixtures("app")
+    def test_return_to_calculated_summary(self):
+        self.schema = load_schema_from_name("test_calculated_summary")
+
+        current_location = Location(
+            section_id="default-section", block_id="fifth-number-block"
+        )
+
+        routing_path = RoutingPath(
+            ["sixth-number-block"],
+            section_id="default-section",
+        )
+        next_location_url = self.router.get_next_location_url(
+            current_location,
+            routing_path,
+            return_to="calculated-summary",
+            return_to_block_id="currency-total-playback-skipped-fourth",
+        )
+        expected_location_url = Location(
+            section_id="default-section",
+            block_id="currency-total-playback-skipped-fourth",
+        ).url()
+
+        assert expected_location_url == next_location_url
+
+    @pytest.mark.usefixtures("app")
+    def test_return_to_calculated_summary_invalid_return_to_block_id(self):
+        self.schema = load_schema_from_name("test_calculated_summary")
+
+        current_location = Location(
+            section_id="default-section", block_id="fifth-number-block"
+        )
+
+        routing_path = RoutingPath(
+            ["sixth-number-block", "fifth-number-block"],
+            section_id="default-section",
+        )
+        next_location_url = self.router.get_next_location_url(
+            current_location,
+            routing_path,
+            return_to="calculated-summary",
+            return_to_block_id="non-valid-block",
+        )
+        expected_location_url = Location(
+            section_id="default-section",
+            block_id="sixth-number-block",
+        ).url()
+
+        assert expected_location_url == next_location_url
+
+    @pytest.mark.usefixtures("app")
+    def test_return_to_calculated_summary_empty_return_to_block_id(self):
```

Comment:
This and `test_return_to_calculated_summary_invalid_return_to_block_id` could be one test and parametrized.
Also, would it not be `None` instead of `""`.?